### ISSUE TITLE
Carthage: 'AppAuth/AppAuth.h' file not found 

### ISSUE THREAD

**Issue by kadikraman** [FormidableLabs/react-native-app-auth#188]
Build fails with `AppAuth/AppAuth.h file not found` when installing AppAuth-iOS from Carthage.

**Xcode version:** 10
**RN version:** 57.2

Also confirmed an issue with Xcode 9.4.1 and RN 0.55.4 so not likely to be a problem with RN, but header search paths.

**Comment by kadikraman** [comment]
<img width="776" alt="screen shot 2018-10-10 at 11 26 18" src="https://user-images.githubusercontent.com/6534400/46730207-5c860500-cc7f-11e8-9021-f1c74e96a816.png">

So in `react-native-app-auth` we're using header search paths to link to `AppAuth-iOS` installed in the user's project:

```
$(SRCROOT)/../../../ios/Pods/AppAuth
$(SRCROOT)/../../../ios/Pods/Headers
```

I'm not very experienced with Carthage, but I think we need to somehow achieve the equivalent.

**Comment by kadikraman** [comment]
@benjarwar Alright, I think I've got it working.

I've added the framework search path to the library in [this PR](https://github.com/FormidableLabs/react-native-app-auth/pull/189) (you can install from PR with `yarn add FormidableLabs/react-native-app-auth#189/head`), but you'll also have to do the following in your own project:

1. Drag and drop the `AppAuth.framework` from `ios/Carthage/Build/iOS` under `Frameworks` in `Xcode`
2. Add a copy files build step for `AppAuth.framework` as specified here https://github.com/Carthage/Carthage/issues/616#issuecomment-121095995

Could you verify that this works for you? Thanks 🙏

**Comment by benjarwar** [comment]
@kadikraman thanks for your attention here! Most of your suggestions are things we've already tried/implemented, unfortunately. Our CI is particularly flaky. The app fails to build about 50% of the time.

We thought [that deleting DerivedData](https://stackoverflow.com/questions/38016143/how-can-i-delete-derived-data-in-xcode-8) solved things locally. But I have at least one team member who reports that hasn't helped :P

Here're more details/questions RE our project's configuration:

- We already have `AppAuth.framework` listed in "Linked Frameworks and Libraries". But I believe we dragged and dropped the `AppAuth.framework` file into the Libraries section, instead of Frameworks. Does this matter?
- We have `$(PROJECT_DIR)/Carthage/Build/iOS` set in our projects Framework Search Paths. I'd think this would cover what you're doing in [your PR](https://github.com/FormidableLabs/react-native-app-auth/pull/189), but maybe there's some conflict/load order issue that makes it preferable to have the path set in RNAppAuth itself?
- While we're not precisely following that [Carthage comment](https://github.com/Carthage/Carthage/issues/616#issuecomment-121095995), we do copy over our frameworks in a similar way:

![screen shot 2018-10-10 at 9 58 12 am](https://user-images.githubusercontent.com/2565278/46741457-0f039b00-cc73-11e8-9ca0-87e27b786e73.png)

**Comment by kadikraman** [comment]
@benjarwar Thanks for the suggestion. I think you're right, we should have better examples, so I've added a new issue for it.

As for the rest of your questions:

**Should it be a framework?** Well, it's named `AppAuth.framework`, so based on that I'm guessing yes? The other things under `Libraries` are all `.xcodeproj`.

Then again, if your app has been working fine so far, perhaps I'm wrong. Xcode has Header, Library and Framework search paths, so maybe if it's linked as a Library, then `$(PROJECT_DIR)/Carthage/Build/iOS` needs to be in your Library search paths?

**Does the `$(PROJECT_DIR)/Carthage/Build/iOS` have to be in the library** or can it be in the project. Yes, I think it does have to be set in the library, because the library has to resolve paths outside of itself, similarly how we had to add the Pods directory in Header search paths. For an empirical example, the way I've been debugging this is I created a brand new RN project, installed `react-native-app-auth` and `AppAuth-iOS` using Carthage. I have `$(PROJECT_DIR)/Carthage/Build/iOS` set in both the project and `$(SRCROOT)/../../../ios/Carthage/build/iOS` in RNAppAuth Framework search paths, but when I delete it from the latter, my Xcode build immediately fails.

**Comment by kadikraman** [comment]
This fix is released in [v4.0.0](https://github.com/FormidableLabs/react-native-app-auth/releases/tag/v4.0.0) along with [many more example apps](https://github.com/FormidableLabs/react-native-app-auth/tree/master/Example), as suggested, including one just for Carthage.
